### ISSUE TITLE
inv:add_item does not work correctly in callbacks that return an itemstack

### ISSUE THREAD

**Issue by Extex101** [minetest/minetest#9190]
##### Minetest version

```
5.1.0
```

##### Summary

When using `inv:add_item(list, item)` if item is to be added to inventory space which is currently selected it won't get added

##### Steps to reproduce

Make a mod with this code

minetest.register_node("mod:block", {
	description = "add_item bug",
	on_rightclick = function(pos, node, player, itemstack, pointed_thing)
		local inv = player:get_inventory()
		inv:add_item("main", ItemStack("default:stick"))
	end,
})

**Comment by SmallJoker** [comment]
Not a new bug. Can reproduce this in 0.4.17.1 as well.

**Comment by Extex101** [comment]
Do you know if there is a hacky way around this bug?

**Comment by SmallJoker** [comment]
```Lua
leftover = itemstack:add_item("default:stick")
if leftover:is_empty() then
	return itemstack
end
inv:add_item("main", ItemStack("default:stick"))
```

**Comment by appgurueu** [comment]
Not a bug IMO. It is to be expected that `inv:add_item` has to be called after the item is replaced, else the changes made can be overwritten. Another "hackaround" would accordingly be wrapping the call in `minetest.after`.

**Comment by sfan5** [comment]
same fundamental problem as #6546, solved by #11662
